core/src/com/unciv/ui/newgamescreen/GameOptionsTable.kt

```diff
@@ -89,23 +92,32 @@ class GameOptionsTable(
                 gameParameters.isOnlineMultiplayer = it
                 updatePlayerPickerTable("")
             }
-    
-    private fun Table.addReligionCheckbox() =
-            addCheckbox("Enable Religion", gameParameters.religionEnabled)
-            { gameParameters.religionEnabled = it }
 
-    private fun Table.addCityStatesSlider() {
-        val numberOfCityStates = ruleset.nations.filter { it.value.isCityState() }.size
-        if (numberOfCityStates == 0) return
+    private fun numberOfCityStates() = ruleset.nations.values.count {
+        it.isCityState() &&
+                (it.cityStateType != CityStateType.Religious || gameParameters.religionEnabled) &&
+                !it.hasUnique(UniqueType.CityStateDeprecated)
+    }
+
+    private fun Table.addReligionCheckbox(cityStateSlider: UncivSlider?) =
+        addCheckbox("Enable Religion", gameParameters.religionEnabled) {
+            gameParameters.religionEnabled = it
+            cityStateSlider?.run { setRange(0f, numberOfCityStates().toFloat()) }
```

Comment:
why .run not .apply?

Reply:
For me, expresses that I'm not using the return value, `Unit` in this case, while `apply` is more like a 'I do something to it - pardon, to a shadowed this - but am not done with it afterwards'... Not really important, taste mostly. Like coming from M$ languages I thought `with` to be a good thing at first, now it's off the palette for me, `run` instead. Or the use of `also` for the sole reason I don't _like_ to use that shadowed `this` explicitly (often meaning on the right side of the expression or as function parameter), but implicit is fine. That's the whole 'why' in a nutshell.